### DISCUSSION TITLE
Appium for PyQT Windows apps automation

### DISCUSSION THREAD

**Discussion by sanchouz12** [appium/appium#20040] [General]
Greetings, I have a question: is Appuim suitable for automating Windows applications, written in PyQT? The app is bundled using PyInstaller for MacOS and Windows. Automation on MacOS works perfectly: I'm able to access all needed attributes of elements and perform all needed actions. But when I tried to automate the same app, but for Windows, the only useful thing I could do is just to find elements, because Appium can't retrieve much information.

Or maybe I have to do more setup, so Appium can detect everything I need? My current flow is:
1. Start Appium server with installed **appium-windows-driver** (and WinAppDriver, of course)
2. Run small script, where I open the app, find element and retrieve all xml into a file

Here's what I get:
<img width="1285" alt="Знімок екрана 2024-04-26 о 12 39 05" src="https://github.com/appium/appium/assets/53550887/14da32c5-64e1-401c-9119-62e0f40cdaff">

Xml of that element:
<img width="816" alt="Знімок екрана 2024-04-26 о 13 10 17" src="https://github.com/appium/appium/assets/53550887/589742e6-4ec9-4e74-8345-e79c52edcf9f">

And that's what I get on Mac:
<img width="773" alt="Знімок екрана 2024-04-26 о 15 40 54" src="https://github.com/appium/appium/assets/53550887/3951878b-5446-474e-8926-8b501e7e5944">
<img width="1527" alt="Знімок екрана 2024-04-26 о 15 38 55" src="https://github.com/appium/appium/assets/53550887/63d1bbb1-032d-4f6a-8b4a-f2484251632c">

Any help much appreciated, thanks

**Comment by jlipps** [comment]
What you get is what you get with WinAppDriver, unfortunately. MS has stopped maintaining it so it's not likely to improve. You could ask around the Appium forums in case anyone has any ideas there though.
